### PR TITLE
Last update from homework week3

### PR DESCRIPTION
**Pull request by malnajar**
Hi ...

i just changed few things ... lifecycle methods, not using the index in my functions, and less components
also i changed the way of importing data from the localStorage 

if there is anything else i can do to make my code better please let me know 

thanks ..

### CODE REVIEW

**Comment by isaachinman** [comment]
Hi @malnajar, good to hear you came back to the homework and did more on it! I see you added some things like `localStorage.js` etc but the codebase seems pretty much the same to me. Can you explain what changes you made?

**Comment by malnajar** [comment]
Sorry but i've uploaded the wrong files ... you can see the changes now

File: todo-app/src/App.js [hunk 1]
```diff
@@ -1,10 +1,98 @@
 import React, { Component } from 'react';
+import Header from './Components/Header'
+import Container from './Components/Container'
+import { LoadFromLocalStorage, SaveToLocalStorage} from './localStorage'
+
+import './App.css';
+import data from './todoList.json';
```
Comment (malnajar) [inline todo-app/src/App.js]: as per the last comment from you on my home work i have changed the way of importing the data

File: todo-app/src/App.js [hunk 2]
```diff
@@ -1,10 +1,98 @@
...
 
 class App extends Component {
+  state = {
+    todoItems: LoadFromLocalStorage() || data
+  }
+
+  componentDidUpdate = () => {
```
Comment (malnajar) [inline todo-app/src/App.js]: here i have removed the lifecycle methods  and used the imported file "localSrotrage.js" and i have changed the state

File: todo-app/src/App.js [hunk 3]
```diff
@@ -1,10 +1,98 @@
...
+    SaveToLocalStorage(this.state.todoItems)
+  }
+
+  handelSelected = (id) => {
```
Comment (malnajar) [inline todo-app/src/App.js]: not using the index anymore.

File: todo-app/src/App.js [hunk 4]
```diff
@@ -1,10 +1,98 @@
...
+    this.setState({ todoItems })
+  }
+
+  handelNewTask = (newTask, newDeadLine) => {
+
+    const todoItems = this.state.todoItems
+    const id = this.state.todoItems.map(task => task.id)
+    const newId = Math.max(...id) + 1
+    todoItems.push({
+      "id": newId,
```
Comment (malnajar) [inline todo-app/src/App.js]: not using the index to implement a new id

File: todo-app/src/App.js [hunk 5]
```diff
@@ -1,10 +1,98 @@
...
+        }
+      }
+    })
+    this.setState({ todoItems })
+  }
+
+  handelDeleteTask = (id, selected) => {
+
+    const todoItems = this.state.todoItems.filter(task => task.id !== id)
+    todoItems.map((item, index) => {
```
Comment (malnajar) [inline todo-app/src/App.js]: it seems that i forgot about this one i will change it and update the file asap

File: todo-app/src/App.js [hunk 6]
```diff
@@ -1,10 +1,98 @@
...
+    const todoItems = this.state.todoItems.map(task => {
+      if (task.id === id) {
+        return {
+          "id": task.id,
+          "description": task.description,
+          "deadline": task.deadline,
+          "selected": !task.selected
+        }
+      } else {
+        return {
```
Comment (isaachinman) [inline todo-app/src/App.js]: Why not just `return task`?

File: todo-app/src/App.js [hunk 7]
```diff
@@ -1,10 +1,98 @@
...
+    const todoItems = this.state.todoItems.map(task => {
+      if (task.id === id) {
+        return {
+          "id": task.id,
+          "description": update,
+          "deadline": task.deadline,
+          "selected": task.selected
+        }
+      } else {
+        return {
```
Comment (isaachinman) [inline todo-app/src/App.js]: Same question here, you should just be able to `return task`.

**Comment by isaachinman** [comment]
@malnajar Please open a new PR for Week 4's HW.

**Pull request closed without merging**